sdk/storage/azblob/service/models.go

```diff
@@ -188,6 +192,43 @@ func (o *SetPropertiesOptions) format() (generated.StorageServiceProperties, *ge
 		return generated.StorageServiceProperties{}, nil
 	}
 
+	defaultVersion := to.Ptr[string]("1.0")
+	defaultAge := to.Ptr[int32](0)
+	emptyStr := to.Ptr[string]("")
+
+	if o.Cors != nil {
+		for i := 0; i < len(o.Cors); i++ {
+			if o.Cors[i].AllowedHeaders == nil {
+				o.Cors[i].AllowedHeaders = emptyStr
```

Comment:
How was this working earlier ?

Reply:
Can you share some reference for this?
If its documented somewhere in the Microsoft docs, then maybe we should also document this.

Reply:
So if you reference the generated docs it clearly says all the fields are required if one is provided. However, the average customer will usually provide the allowedOrigins and allowedHeaders, the other parameters are not as intiutitive to keep as an empty str or set to 0, so I added these checks as quick validations. You can see this in python as well here https://github.com/Azure/azure-sdk-for-python/blob/8a4472205bf4fff0566a427e41888d0c0dc0fecf/sdk/storage/azure-storage-blob/azure/storage/blob/_models.py#L333-L335